src/modules/internet/libcurl.c

```diff
@@ -236,35 +236,25 @@ static void curlCommon(CURL *hnd, int redirect, int verify)
 	curl_easy_setopt(hnd, CURLOPT_SSL_VERIFYHOST, 0L);
 	curl_easy_setopt(hnd, CURLOPT_SSL_VERIFYPEER, 0L);
     }
-#if 0
     // for consistency, but all utils:::makeUserAgent does is look up an option.
     SEXP sMakeUserAgent = install("makeUserAgent");
     SEXP agentFun = PROTECT(lang2(sMakeUserAgent, ScalarLogical(0)));
     SEXP utilsNS = PROTECT(R_FindNamespace(mkString("utils")));
     SEXP sua = eval(agentFun, utilsNS);
     UNPROTECT(1); /* utilsNS */
     PROTECT(sua);
-    if(TYPEOF(sua) != NILSXP)
-	curl_easy_setopt(hnd, CURLOPT_USERAGENT, CHAR(STRING_ELT(sua, 0)));
+
+    /* Prefix with libcurl version, always */
```

Comment:
Before, you could use a custom `HTTPUserAgent` string as-is, as long as it didn't match the R default. But now it'll always include a libcurl string, even if you wanted to remove your user agent. Would it be worth preserving the previous behavior for custom user agents?

Reply:
By previous behavior you mean to prefix only when the user agent string starts with `R `? I don't know, that seems even more ad hoc to me.

Reply:
Oh no, I mean only prefix if the user agent fully matches the default `R (<version> <platform> <arch> <os>)`. I don't think I explained it well, so here are some examples. This table shows the `HTTPUserAgent` option and the libcurl user agents for the previous (R 3.6) behavior, the new behavior, and what I was expecting:

|`HTTPUserAgent`|Previous (R 3.6)|New|Expected|
|---------------|--------|---|---------|
|`NULL`|`libcurl/7.29.0`|`libcurl/7.29.0`|no user agent|
|`""`|no user agent|`libcurl/7.29.0`|no user agent|
|`"custom"`|`custom`|`libcurl/7.29.0, custom`|`custom`|
|`"R (custom)"`|`libcurl/7.29.0`|`libcurl/7.29.0, R (custom)`|`R (custom)`|
|default|`libcurl/7.29.0`|`libcurl/7.29.0, R (3.7.0 x86_64-pc-linux-gnu x86_64 linux-gnu)`|`libcurl/7.29.0, R (3.7.0 x86_64-pc-linux-gnu x86_64 linux-gnu)`|

But if there's no easy way to detect a custom user agent (as you mentioned in the email thread), then I think the current behavior to always prefix libcurl is fine.

Side note - I didn't realize before, but the previous (R 3.6) behavior for `NULL` actually seems broken? I thought there would be no user agent set based on the docs:
> HTTPUserAgent:
> string used as the ‘user agent’ in HTTP(S) requests by download.file, url and curlGetHeaders, or **NULL when requests will be made without a user agent header.**

If it's not feasible to detect a custom `HTTPUserAgent`, perhaps we could detect if it's NULL, and exclude the user agent header it if so?

Reply:
See below.